File: AdventOfCode/2022/02/rps.py
"""
- part 1
rock = A , X , points = 1
paper = B , Y , points = 2
scissors = C , Z , points = 3

win = 6, draw = 3, lost = 0
point per rount = outcome + what you used
"""

#part1
with open("AdventOfCode\\2022\\02\\input.txt") as f:
    points = 0
    for game in f:
        elf, you = game.strip().split(" ")
        if elf == "A" and you == "Y":
            points += 2 + 6
        elif elf == "A" and you == "Z":
            points += 3 + 0
        elif elf == "A" and you == "X":
            points += 1 + 3
        elif elf == "B" and you == "Y":
            points += 2 + 3
        elif elf == "B" and you == "Z":
            points += 3 + 6
        elif elf == "B" and you == "X":
            points += 1 + 0
        elif elf == "C" and you == "Y":
            points += 2 + 0
        elif elf == "C" and you == "Z":
            points += 3 + 3
        elif elf == "C" and you == "X":
            points += 1 + 6
print(points)

"""
- part 2
X = need to lose
Y = need to draw
Z = need to win
"""

#part2
with open("AdventOfCode\\2022\\02\\input.txt") as f:
    points = 0
    for game in f:
        elf, outcome = game.strip().split(" ")
        if elf == "A" and outcome == "Y":
            points += 1 + 3
        elif elf == "A" and outcome == "Z":
            points += 2 + 6
        elif elf == "A" and outcome == "X":
            points += 3 + 0
        elif elf == "B" and outcome == "Y":
            points += 2 + 3
        elif elf == "B" and outcome == "Z":
            points += 3 + 6
        elif elf == "B" and outcome == "X":
            points += 1 + 0
        elif elf == "C" and outcome == "Y":
            points += 3 + 3
        elif elf == "C" and outcome == "Z":
            points += 1 + 6
        elif elf == "C" and outcome == "X":
            points += 2 + 0
print(points)

File: AdventOfCode/2022/02/rps.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main(){
    //reading file 
    fstream file;
    file.open("C:\\Users\\matej\\Documents\\Projects\\AdventOfCode\\2022\\02\\input.txt", ios::in);
    int points_1 = 0;
    int points_2 = 0;

    if (file.is_open()){
        string line, elf, you;
        
        while (getline(file,line)){
            elf = line[0];
            you = line[2];
            
            //part1
            if (elf == "A" && you == "Y"){
                points_1 += (2 + 6);
            }else if (elf == "A" && you == "Z"){
                points_1 += (3 + 0);
            }else if (elf == "A" && you == "X"){
                points_1 += (1 + 3);
            }else if (elf == "B" && you == "Y"){
                points_1 += (2 + 3);
            }else if (elf == "B" && you == "Z"){
                points_1 += (3 + 6);
            }else if (elf == "B" && you == "X"){
                points_1 += (1 + 0);
            }else if (elf == "C" && you == "Y"){
                points_1 += (2 + 0);
            }else if (elf == "C" && you == "Z"){
                points_1 += (3 + 3);
            }else if (elf == "C" && you == "X"){
                points_1 += (1 + 6);
            }

            //part2
            if (elf == "A" && you == "Y"){
                points_2 += (1 + 3);
            }else if (elf == "A" && you == "Z"){
                points_2 += (2 + 6);
            }else if (elf == "A" && you == "X"){
                points_2 += (3 + 0);
            }else if (elf == "B" && you == "Y"){
                points_2 += (2 + 3);
            }else if (elf == "B" && you == "Z"){
                points_2 += (3 + 6);
            }else if (elf == "B" && you == "X"){
                points_2 += (1 + 0);
            }else if (elf == "C" && you == "Y"){
                points_2 += (3 + 3);
            }else if (elf == "C" && you == "Z"){
                points_2 += (1 + 6);
            }else if (elf == "C" && you == "X"){
                points_2 += (2 + 0);
            }
        }
    }
    file.close();

    cout << points_1 << endl;
    cout << points_2 << endl;

    return 0;
}
